code/controllers/subsystems/documents.dm

```diff
@@ -0,0 +1,242 @@
+				add_document(
+					document_query.item[1],
+					document_query.item[2],
+					document_query.item[3],
+					document_query.item[4],
+					json_decode(document_query.item[5]))
+			catch(var/exception/ec)
+				log_debug("SSdocs: Error when loading document: [ec]")
+
+//Loads the document data from JSON
+/datum/controller/subsystem/docs/proc/load_from_json()
+	var/list/docsconfig = list()
+	try
+		docsconfig = json_decode(return_file_text("config/docs.json"))
+	catch(var/exception/ej)
+		log_debug("SSdocs: Warning: Could not load config, as docs.json is missing - [ej]")
+		return
+
+	//Reset the currently loaded data
+	reset_docs()
+
+	//Load the documents
+	for (var/document in docsconfig)
+		CHECK_TICK
+		try
+			log_ss("docs", "Tags (JSON): [list2params(docsconfig[document]["tags"])]")
+			add_document(
+				docsconfig[document]["name"],
+				docsconfig[document]["title"],
+				docsconfig[document]["chance"],
+				docsconfig[document]["content"],
+				docsconfig[document]["tags"])
+		catch(var/exception/ec)
+			log_ss("docs","Error when loading document: [ec]")
+			log_debug("SSdocs: Error when loading document: [ec]")
+	return 1
+
+/datum/controller/subsystem/docs/proc/add_document(var/name,var/title,var/chance,var/content,var/tags)
+	var/datum/docs_document/dd = new()
+	dd.name = name
+	dd.title = title
+	dd.chance = chance
+	dd.content = content
+	if(istext(tags))
+		tags = splittext(tags, ";")
+	dd.tags = tags
+
+	//Adds the document to the docs, sorted by tags
+	for(var/t in tags)
+		if(!(docs_by_tags[t]))
+			docs_by_tags[t] = list()
+			total_by_tags[t] = 0
+		docs_by_tags[t][dd.name] += dd
+		total_by_tags[t] += dd.chance
+
+	//Add the document to the docs list
+	docs[dd.name] = dd
+	src.total_docs += dd.chance
+	return dd
+
+/datum/docs_document
+	var/name = "document" // internal name of the document
+	var/title = "paper" // player-facing title of the document
+	var/chance = 0
+	var/content = "" // Can contain html, but pencode is not processed.
+	var/list/tags = list() // A list of tags that apply to the document.
+
+/obj/random/document
+	name = "random document"
+	desc = "This is a random text document."
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "paper"
+	has_postspawn = 1
+	var/tags = list() // for use by mappers, typically
+
+/obj/random/document/item_to_spawn()
+	return /obj/item/weapon/paper
+
+/obj/random/document/post_spawn(var/obj/item/spawned)
+	var/list/total_tags = src.tags | list(SSDOCS_MEDIUM_PAPER)
+	log_ss("docs","Tags selected: [list2params(total_tags)]")
+	var/datum/docs_document/doc
+	if(total_tags.len == 1)
+		doc = SSdocs.pick_document_by_tag(total_tags[1])
+		if(!istype(doc))
+			log_ss("docs","pick_document_by_tag returned null paper!")
+			return
+	else
+		doc = SSdocs.pick_document_by_tags(total_tags)
+		if(!istype(doc))
+			log_ss("docs","pick_document_by_tags returned null paper!")
+			return
+	if(!istype(spawned, /obj/item/weapon/paper))
```

Comment:
Since `spawned` _has_ to be paper for the proc to function, please path it as such in the proc prototype:
```dm
/obj/random/document/post_spawn(/obj/item/weapon/paper/spawned)
```

This also allows you to shorten the type check:
`if (!istype(spawned))`

And finally. Argument validation should be done at the top of the function. This makes it easier to read and saves us some arbitrary amount of work upon a bad argument